src/sagemaker_xgboost_container/algorithm_mode/train.py

```diff
@@ -28,10 +29,31 @@
 from sagemaker_xgboost_container.callback import add_debugging
 from sagemaker_xgboost_container.constants.xgb_constants import CUSTOMER_ERRORS
 
+MODEL_NAME = "xgboost-model"
 
 logger = logging.getLogger(__name__)
 
 
+def add_sigterm_handler(model_dir, is_master):
+    """Stop training and cleanup model directory when SIGTERM is received.
+
+    Model directory is only cleaned if is_master is True. Otherwise program terminates.
+
+    :param model_dir: Directory where model is saved
+    :param is_master: True if single node training, or the current node is the master node in distributed training
+    """
+    def _terminate():
+        os._exit(0)
+
+    def _cleanup_files(signo, frame):
+        if is_master:
+            train_utils.cleanup_dir(model_dir, MODEL_NAME)
```

Comment:
Cleanup is required because the program can be terminated during `save_intermediate_model` callback is running, which can leave `tmp` files behind in the directory. 
For the second part, currently our model loading side assumes there is only one file in the `/opt/ml/model` directory.